extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -175,9 +175,10 @@ private Uni<SecurityIdentity> processRedirectFromOidc(RoutingContext context, Oi
         String[] parsedStateCookieValue = COOKIE_PATTERN.split(stateCookie.getValue());
         OidcUtils.removeCookie(context, oidcTenantConfig, stateCookie.getName());
         if (!parsedStateCookieValue[0].equals(stateQueryParam.get(0))) {
-            LOG.debug("State cookie value does not match the state query parameter value, "
-                    + "completing the code flow with HTTP status 401");
-            return Uni.createFrom().failure(new AuthenticationCompletionException());
+            final String error = "State cookie value does not match the state query parameter value, "
```

Comment:
I'd like to understand if this can happen in normal situation.

For example I open all the new links in new tabs because I simply don't trust SPAs they are well written and I prefer to keep the tab I am working with intact. But parallel code flows are not supported. Can this happen that state cookie value doesn't match the state query parameter because I work in multiple tabs? Because I don't think it would be useful for administrators to read this particualr scenario.

Reply:
@michalvavrik This is actually covers a case where a parallel tab login is now allowed...

But to be honest, I'm not sure why we are looking at individual log messages. The idea of the PR is to get 401 related errors returned to users in devmode.

Reply:
> But to be honest, I'm not sure why we are looking at individual log messages. The idea of the PR is to get 401 related errors returned to users in devmode.

You have changed logging level from "enabled when I am debugging" to enabled always (because everyone is logging errors). Our users pay for these logs and logging is slowing down things, so I am trying to figure if they actually will find these messages useful in production. You said yes and I believe you.

Reply:
@michalvavrik I see what I mean, it is just that 401 deserves a more visible kind of message, we have plenty of debug messages which will stay at the debug level.

In general, I'd like to make sure that when the user gets 401, for example, in prod, the teams don't scratch their heads why the user got 401 in that case, we are keeping getting messages that it is not easy to find out why the failure occurred

Hope it makes sense

Reply:
> In general, I'd like to make sure that when the user gets 401, for example, in prod, the teams don't scratch their heads why the user got 401 in that case, we are keeping getting messages that it is not easy to find out why the failure occurred

Yes, I think you confirm what I wanted to know. That when 401 occurrences, they can use logged information to fix issue, not just to be informed that something regular happened. I tried to figure from flow of the code that leads to these messages, but it's not always straightforward.

Thanks, please consider this thread closed.